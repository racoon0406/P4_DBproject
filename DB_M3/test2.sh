#!/bin/bash
# test2: tests range and select requests with versioning and 
#        ACL behavior for clients with different key range access

#          read      write
# Alice  [0, 1024]  [0, 512]
# Bob    [0, 256]   [0, 256]

# Alice range

# keys in [0, 4] second version
./send_Alice.py range 0 4 1

# keys in [0, 30] first version
./send_Alice.py range 1 30 0

# keys in [512, 513] first version
./send_Alice.py range 512 513 0

# keys in [1022, 1024] first version
./send_Alice.py range 1022 1024 0


# Alice select

# >
./send_Alice.py select gt 1022 0

# >=
./send_Alice.py select gteq 1022 0

# <
./send_Alice.py select ls 22 0
./send_Alice.py select ls 22 1

# <=
./send_Alice.py select lseq 22 0
./send_Alice.py select lseq 22 1

# ==
./send_Alice.py select eq 512 1
./send_Alice.py select eq 513 0
./send_Alice.py select eq 1024 0


# Bob range

# keys in [0, 4] second version
./send_Bob.py range 0 4 1

# keys in [0, 30] first version
./send_Bob.py range 1 30 0

# keys in [255, 258] first version
./send_Bob.py range 255 258 0     # Bob has no read access for [257, 258]

# keys in [1022, 1024] first version
./send_Bob.py range 1022 1024 0   # Bob has no read access


# Bob select

./send_Bob.py select gt 1022 0    # Bob has no read access

# >=
./send_Bob.py select gteq 1022 0  # Bob has no read access

# <
./send_Bob.py select ls 22 0
./send_Bob.py select ls 22 1

# <=
./send_Bob.py select lseq 22 0
./send_Bob.py select lseq 22 1

# ==
./send_Bob.py select eq 257 0     # Bob has no read access
./send_Bob.py select eq 258 0     # Bob has no read access
./send_Bob.py select eq 255 0
./send_Bob.py select eq 256 0
./send_Bob.py select eq 256 1