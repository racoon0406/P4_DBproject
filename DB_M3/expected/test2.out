sniffing on eth0
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 21, 11, 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 40, 30, 20, 10]
[Client Alice]: get multiple values from switch 1
[200, 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
received a PONG from switch 2
received a PONG from switch 1
[Client Alice]: get multiple values from switch 1
[300, 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 220, 210]
[Client Alice]: get multiple values from switch 1
['NULL', 51200]
[Client Alice]: get multiple values from switch 2
['NULL', 'NULL', 'NULL']
[Client Alice]: get multiple values from switch 2
['NULL', 'NULL']
[Client Alice]: get multiple values from switch 2
['NULL', 'NULL', 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 40, 30, 20, 10, 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Alice]: get multiple values from switch 1
[210, 200]
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 21, 11, 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 201]
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 40, 30, 20, 10, 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Alice]: get multiple values from switch 1
[220, 210, 200]
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 21, 11, 'NULL']
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
received a PONG from switch 2
received a PONG from switch 1
[Client Alice]: get multiple values from switch 1
['NULL', 'NULL', 201]
[Client Alice]: get multiple values from switch 1
[51201]
[Client Alice]: get multiple values from switch 2
['NULL']
[Client Alice]: get multiple values from switch 2
['NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 21, 11, 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 40, 30, 20, 10]
[Client Bob]: get multiple values from switch 1
[200, 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Bob]: get multiple values from switch 1
[300, 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 220, 210]
Error!!! Client Bob has no access to read the key(s)
Error!!! Client Bob has no access to read the key(s)
Error!!! Client Bob has no access to read the key(s)
Error!!! Client Bob has no access to read the key(s)
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 40, 30, 20, 10, 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Bob]: get multiple values from switch 1
[210, 200]
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 21, 11, 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 201]
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 40, 30, 20, 10, 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
received a PONG from switch 2
received a PONG from switch 1
[Client Bob]: get multiple values from switch 1
[220, 210, 200]
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 21, 11, 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL', 'NULL']
[Client Bob]: get multiple values from switch 1
['NULL', 'NULL', 201]
Error!!! Client Bob has no access to read the key(s)
Error!!! Client Bob has no access to read the key(s)
[Client Bob]: get multiple values from switch 1
[25500]
[Client Bob]: get multiple values from switch 1
[25600]
[Client Bob]: get multiple values from switch 1
[25601]
